include/amici_model.h

```diff
@@ -300,30 +312,282 @@ namespace amici {
         
         void fdJzdx(std::vector<realtype> *dJzdx, const int nroots, realtype t, const ExpData *edata, const ReturnData *rdata);
         
-        void initialize(AmiVector *x, AmiVector *dx, const UserData *udata);
+        void initialize(AmiVector *x, AmiVector *dx);
         
-        void initializeStates(AmiVector *x, const UserData *udata);
+        void initializeStates(AmiVector *x);
         
-        void initHeaviside(AmiVector *x, AmiVector *dx, const UserData *udata);
+        void initHeaviside(AmiVector *x, AmiVector *dx);
         
         /** number of paramaeters wrt to which sensitivities are computed
          * @return length of sensitivity index vector
          */
         const int nplist() const {
-            return plist.size();
-        };
+            return plist_.size();
+        }
         /** total number of model parameters
          * @return length of parameter vector
          */
         const int np() const {
-            return p.size();
-        };
+            return originalParameters.size();
+        }
         /** number of constants
          * @return length of constant vector
          */
         const int nk() const {
-            return k.size();
-        };
+            return k_.size();
+        }
+
+        /** constants
+         * @return pointer to constants array
+         */
+        const double *k() const{
+            return k_.data();
+        }
+
+        /**
+         * @brief Get nmaxevent
+         * @return nmaxevent
+         */
+        int nMaxEvent() const {
+            return nmaxevent;
+        }
+
+        /**
+         * @brief setNMaxEvent
+         * @param nmaxevent
+         */
+        void setNMaxEvent(int nmaxevent) {
+            this->nmaxevent = nmaxevent;
+        }
+
+        /**
+         * @brief Get number of timepoints
+         * @return number of timepoints
+         */
+        int nt() const {
+            return ts.size();
+        }
+
+        /**
+         * @brief getParameterScale
+         * @return
+         */
+        AMICI_parameter_scaling getParameterScale() const {
+            return pscale;
+        }
+
+        /**
+         * @brief setParameterScale
+         * @param pscale
+         */
+        void setParameterScale(AMICI_parameter_scaling pscale) {
+            this->pscale = pscale;
+            unscaledParameters.resize(originalParameters.size());
+            unscaleParameters(unscaledParameters.data());
+        }
+
+        /**
+         * @brief getPositivityFlag
+         * @return
+         */
+        std::vector<int> getPositivityFlag() const {
+            return qpositivex;
+        }
+
+        /**
+         * @brief setPositivityFlag
+         * @param qpositivex
+         */
+        void setPositivityFlag(std::vector<int> const& qpositivex) {
+            if(qpositivex.size() != (unsigned) this->nx)
+                throw AmiException("Dimension mismatch.");
```

Comment:
can we put more informative error messages here? the text in the AmiException will be printed as error message.